File: HackerRank Solutions/Algorithms/Designer PDF Viewer/Designer PDF Viewer.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t=26;
    vector <int>h(t);
    for (int i=0; i<t; i++) {
        cin >> h[i];
    }
    string s;
    cin >> s;
    int height = 0;
    for(int j=0; j<s.length(); j++){
        height = max(height, h[s[j] - 'a']);
    }
    int area = height * 1 * s.length();
    cout << area << endl;
    return 0;
}

File: HackerRank Solutions/Algorithms/Designer PDF Viewer.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the designerPdfViewer function below.
def designerPdfViewer(h, word):
    height = 0
    for i in range(len(word)):
        height = max(height, h[ord(word[i]) - ord('a')])
    area = height * 1 * len(word)
    return(area)
if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    h = list(map(int, input().rstrip().split()))

    word = input()

    result = designerPdfViewer(h, word)

    fptr.write(str(result) + '\n')

    fptr.close()
